### PR TITLE
Remove deprecated `NewWithWriter()`.

### PR DESCRIPTION
**Pull request by dhermes**
In the process, also getting `actions.go` and `command.go` to 100% test coverage.

### CODE REVIEW

File: actions.go [hunk 1]
```diff
@@ -25,6 +25,11 @@ import (
 	"strings"
 )
 
+var (
+	// osExit allows `os.Exit()` to be stubbed during testing.
+	osExit = os.Exit
```
Comment (dhermes) [inline actions.go:30]: This change is "necessary" to get test coverage for the `os.Exit()` call in `Fatalf()`.

However, the "uncertainty" of a var here may not be worth it to you, so I am happy to revert this change.

Going in the **OTHER** direction, we could also put an `ExitFunc` on `Action` the same way we did for `GetenvFunc`.

File: command_test.go [hunk 2]
```diff
@@ -24,13 +24,22 @@ func TestCommand_String(t *testing.T) {
...
 	if got, want := cmd.String(), "::foo::bar"; got != want {
 		t.Errorf("expected %q to be %q", got, want)
 	}
 
-	cmd.Properties = CommandProperties{"bar": "foo"}
+	cmd = Command{
+		Name:       "foo",
+		Message:    "bar",
+		Properties: CommandProperties{"bar": "foo"},
+	}
```
Comment (dhermes) [inline command_test.go:36]: I put it in the commit message, but this change (and the one on line 27) is just a nit by me. I figured "new `Command{}` literal" is slightly better than "modify existing `Command` object" because it visually "localizes" the value being tested.

Happy to revert my editorializing though. (The real goal of the edits was to test the unset `Name` on line 41.)

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+coverage.out
```
Comment (dhermes) [inline .gitignore:1]: I added this since I was generating coverage via

```
rm -f coverage.out
go test -coverprofile=coverage.out ./...
go tool cover -html=coverage.out
```

**Comment by sethvargo** [comment]
Hey @dhermes 

Thank you for the PR. Can we separate the code coverage from removing the function? I want to have a few releases with the deprecated function before removing it.

**Comment by dhermes** [comment]
@sethvargo I pulled out the test-related stuff in #22. Still a design decision to make sure around `os.Exit()` / possibly adding an `ExitFunc` field to `Action`.
